### PR TITLE
Segmentation api deployment

### PR DESCRIPTION
**Pull request by jsharma-cn**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+departuresBuild(project: "segmentation-api") {
+  withJava(version: "java-1.8.0-openjdk") {
```
Comment (aldwinb) [inline Jenkinsfile]: I believe that these lines should be replaced with this: `CnSpringBootBuild( project: "segmentation-api", jdkVersion: "java-1.8.0-openjdk" ) {`. The Java projects that are in Departures use this. One example: https://github.com/CondeNast/booking/blob/bc11004b8c10cefe832ed59117bd1f97c4d21e9c/Jenkinsfile#L1-L10

File: src/main/java/com/condenast/segmentation/config/SwaggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	
+	@Override
+	public void addResourceHandlers(ResourceHandlerRegistry registry) {
+	    registry.addResourceHandler("swagger-ui.html")
+	      .addResourceLocations("classpath:/META-INF/resources/");
+	 
+	    registry.addResourceHandler("/webjars/**")
+	      .addResourceLocations("classpath:/META-INF/resources/webjars/");
+	}
+}
```
Comment (aldwinb) [inline src/main/java/com/condenast/segmentation/config/SwaggerConfig.java]: Can we display the documentation when accessing the home site (e.g. http://localhost:8090)? Right now, it doesn't display anything. Also, I believe our standard is to use ReDoc for the documentation's UI. We can address the ReDoc issue in another PR, but let's fix the home site issue.

Comment (jsharma-cn) [reply src/main/java/com/condenast/segmentation/config/SwaggerConfig.java]: Fixed. Redirected the Rest API docs from home site. I will check ReDoc for the documentation's UI later.

File: Jenkinsfile (hunk 1 above)
Comment (jsharma-cn) [reply Jenkinsfile]: Changed to CnSpringBootBuild.

File: src/main/java/com/condenast/segmentation/config/SwaggerConfig.java (hunk 2 above)
Comment (aldwinb) [reply src/main/java/com/condenast/segmentation/config/SwaggerConfig.java]: Cool! I think we should put the ReDoc changes into another PR. This one is getting huge already  🙂

**Pull request merged**